esrally/driver/runner.py

```diff
@@ -2437,6 +2438,41 @@ def __repr__(self, *args, **kwargs):
         return "delete-ilm-policy"
 
 
+class Sql(Runner):
+    """
+    Executes an SQL query and optionally paginates through subsequent pages.
+    """
+
+    async def __call__(self, es, params):
+        body = mandatory(params, "body", self)
+        mandatory(body, "query", str(self) + ".body")
```

Comment:
This is interesting. I'm not sure that we have another case of a nested parameter that's mandatory. This function call will wind up surfacing an error message like this if the `body` doesn't contain a `query` key:

```
Parameter source for operation 'sql.body' did not provide the mandatory parameter 'query'.
```
That's a little confusing. `query` is a required key nested under the required `body` param, so we'd want the error message to make that clear and say something like:

```
Parameter source for operation 'sql' did not provide the mandatory parameter 'body.query'.
```
The `mandatory` function can't really express this, since it only tests for the existence of of a top-level key. I'd suggest that we do the check here instead like this:

```suggestion
        if body.get("query") is None:
            raise exceptions.DataError(
                f"Parameter source for operation 'sql' did not provide the mandatory parameter 'body.query'. "
                f"Add it to your parameter source and try again."
            )
```